robot_controllers_interface/include/robot_controllers_interface/gyro_handle.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020, Fetch Robotics Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Fetch Robotics Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL FETCH ROBOTICS INC. BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// Author: Vincenzo Barbato
+
+#ifndef ROBOT_CONTROLLERS_INTERFACE_GYRO_HANDLE_H
+#define ROBOT_CONTROLLERS_INTERFACE_GYRO_HANDLE_H
+
+#include <robot_controllers_interface/handle.h>
+#include <geometry_msgs/Vector3.h>
+
+namespace robot_controllers
+{
+
+/**
+ * @brief Base class for a gyro handle used to provide controller 
+ *        direct access to gyro data though controller interface.
+ */
+class GyroHandle : public Handle
+{
+public:
+  GyroHandle()
+  {
+  }
+
+  /** @brief Ensure proper cleanup with virtual destructor. */
+  virtual ~GyroHandle()
+  {
+  }
+
+  /** @brief Get the angular velocities from the gyro */
+  virtual const geometry_msgs::Vector3& getAngularVelocities()const = 0;
```

Comment:
For all function return a value instead of a refererence.    This is less efficient, but forcing the interface to return a reference means that it must store the values internally (which most implementations probably will) but not always.

Reply:
Also remove const at end of function.   Don't force the functions to be const either.

Reply:
One question that I have is should e assume that angular velocities always be `geometry_msgs::Vector3` or should we split them up into the floats like we have for the joint ?

Reply:
Yeah, it looks like this package doesn't depend on geometry messages.  Maybe it is better to split things up.

Reply:
generally I tried to keep everything as ROS-agnostic as possible, so yeah, use floats instead of messages.